lib/interfaces/tdd.js

```diff
@@ -84,6 +86,15 @@ module.exports = function (suite) {
       if (suite.isPending()) {
         fn = null;
       }
+
+      // Check for nested test registration
+      var currentRunnable = suite.rootSuite().currentRunnable;
+      if (currentRunnable) {
+        throw createUnsupportedError(
+          'Nested test "' + title + '" detected inside "' + currentRunnable.title + '".'
+        );
+      }
```

Comment:
[Refactor] This TDD error has the same text as the BDD one. Have you considered making a shared function function for at least the error throwing? Maybe even the checking of the current runnable?

I'm on the fence here and don't mind either way. Personally I normally try to deduplicate when something is repeated 3x. This is only 2x. 🤷 whatever you prefer!

Reply:
I agree, I'll pull this out into a separate function in the file common.js, which contains code that is common to both bdd.js and tdd.js.